### DISCUSSION TITLE
Is there a way to have PR diff code coverage report working with Azure DevOps?

### DISCUSSION THREAD

**Discussion by tonyma** [danielpalme/ReportGenerator#586] [Q&A] [answered]
I am reading from Microsoft doc here: https://learn.microsoft.com/en-us/azure/devops/pipelines/test/codecoverage-for-pullrequests?view=azure-devops

Cannot make it work. Not sure if ReportGenerator can produce such a report with DevOps

**Comment by danielpalme** [comment, marked as answer]
You can archive something similar, but it also requires some setup.
To make coverage history available to _ReportGenerator_, you can use the following setup:
https://github.com/danielpalme/ReportGenerator/wiki/Integration#history

Then you could generate e.g. one of the following reports:
- MarkdownDeltaSummary ([sample](https://reportgenerator.io/resources/reports/MarkdownDeltaSummary/DeltaSummary.md))
- TextDeltaSummary ([sample](https://reportgenerator.io/resources/reports/TextDeltaSummary/DeltaSummary.txt))

Then you could publish the resulting file as a build artifact.

**Reply by tonyma** [reply]
Thanks 👍
